boot-mongodb-elasticsearch/src/test/java/com/example/mongoes/web/controller/SearchControllerTest.java

```diff
@@ -0,0 +1,312 @@
+package com.example.mongoes.web.controller;
+
+import static org.mockito.BDDMockito.given;
+
+import com.example.mongoes.response.AggregationSearchResponse;
+import com.example.mongoes.web.service.SearchService;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
+import org.springframework.test.context.bean.override.mockito.MockitoBean;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@WebFluxTest(SearchController.class)
+class SearchControllerValidationTest {
```

Comment:
@coderabbitai, can you please help me to write integration tests for all endpoints in SearchController